### PR TITLE
Create a FixturesTrait with all necessary methods to interact with fixtures

### PR DESCRIPTION
**Pull request by phenaproxima**
I think it would make sense to have a trait that contains all methods that tests need to interact with and assert the validity of our test fixtures. This could replace parts of UtilsTrait, and totally replace MemoryStorageLoaderTrait.

### CODE REVIEW

File: tests/TestHelpers/FixturesTrait.php [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+     * @param ?int[] $expectedVersions
+     *   The expected versions. The keys are the file names, without the .json
+     *   suffix, and the values are the expected version numbers, or NULL if
+     *   the file should not be present.
+     * @param \ArrayAccess $storage
+     *   The durable storage for the metadata.
+     *
+     * @return void
+     */
+    protected function assertMetadataVersions(array $expectedVersions, \ArrayAccess $storage): void
```
Comment (tedbow) [inline tests/TestHelpers/FixturesTrait.php:176]: could be static

File: tests/TestHelpers/FixturesTrait.php [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+     * @param string $fixtureName
+     *     The name of the fixture to use.
+     * @param string $path
+     *     The relative path (within the fixtures directory) for the client
+     *     data.
+     *
+     * @return MemoryStorage
+     *     Memory storage containing the test client data.
+     */
+    private function loadFixtureIntoMemory(string $fixtureName, string $path = 'client/metadata/current'): MemoryStorage
```
Comment (tedbow) [inline tests/TestHelpers/FixturesTrait.php:118]: could be static

**Review by tedbow — CHANGES_REQUESTED** [review]
I like the trait. a couple nits

File: tests/TestHelpers/FixturesTrait.php (hunk 1 above)
Comment (phenaproxima) [reply tests/TestHelpers/FixturesTrait.php]: Fixed.

File: tests/TestHelpers/FixturesTrait.php (hunk 2 above)
Comment (phenaproxima) [reply tests/TestHelpers/FixturesTrait.php]: Fixed.

**Review by tedbow — APPROVED** [review]
👍

**Pull request merged**